app.rb

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env ruby
+
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+require_relative './rental'
+require_relative './classroom'
+
+class App
+  attr_accessor :books, :people, :rentals, :classroom
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+    @classroom = ['Full-stack Development']
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "Name: \"#{person.name}\", Age: #{person.age}, ID: #{person.id}, Class: #{person.class}"
+    end
+  end
+
+  def create_person
+    puts 'What do you want to create, a Teacher (T) or a Student (S)? [Enter the Letter]:'
+    input_type = gets.chomp.upcase
+    case input_type
+    when 'T'
+      create_teacher
+    when 'S'
+      create_student
+    else
+      invalid_input
+      create_person
+    end
+  end
+
+  def create_teacher
+    puts 'Name:'
+    name = gets.chomp.to_s
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Specialization:'
+    specialization = gets.chomp.to_s
+
+    @people << Teacher.new(specialization, name, age)
+    puts 'Teacher successfully created!'
+  end
+
+  def create_student
+    puts 'Name: '
+    name = gets.chomp.to_s
+    puts 'Age: '
+    age = gets.chomp.to_i
+    puts 'Has parent permission? [Y/N]: '
+    has_permission = gets.chomp.to_s.upcase
+    case has_permission
+    when 'Y'
+      permission = true
+    when 'N'
+      permission = false
+    else
+      invalid_input
+      create_student
+    end
+
+    @people << Student.new(classroom, age, name, parent_permission: permission)
+    puts 'Student created successfully!'
+  end
+
+  def create_book
+    puts 'Title: '
+    title = gets.chomp.to_s
+    puts 'Author: '
+    author = gets.chomp.to_s
+    @books << Book.new(title, author)
+    puts 'Book successfully created'
+  end
+
+  def create_rental
+    puts 'Choose a book from the following list using the numbers beside it'
+    books.each_with_index do |index, book|
+      puts "#{index} Title: #{book.title}, Author #{book.author}"
+    end
+    book_index = gets.chomp.to_i
+    puts 'Select a person that wants to rent a book (choose a number)'
+    people.each_with_index do |index, person|
+      puts "#{index} [#{person.class}] Name: #{person.name}, Age: #{person.age}, ID: #{person.id}"
+    end
+    person_index = gets.chomp.to_i
+    puts 'Enter Date: '
+    date = gets.chomp.to_s
+    rentals << Rental.new(date, people[person_index], books[book_index])
+    puts 'Book successfully rented'
+  end
```

Comment:
- [ ] As I was testing the other features, a book can be created but when I try creating a rental, the app crashes. Please fix the issue on that so a rental is created 👍🏽

<img width="1237" alt="Screenshot 2023-03-17 at 10 04 45 PM" src="https://user-images.githubusercontent.com/50941074/226062315-4d4c67e5-80c9-478f-abe2-af799a491fe6.png">

Reply:
Hello @yetemegn-telaye ,

I encountered a `noMethodError` while trying to create a rental. The `name` method could not be identified in the `create_rental` method. Let's fix that 👍🏽 

<img width="900" alt="Screenshot 2023-03-18 at 15 15 19" src="https://user-images.githubusercontent.com/58692590/226111856-1e5612fb-902f-4fd2-b38b-e6f6151336d0.png">

Reply:
Hi @yetemegn-telaye 
- Great job resolving this 👍🏽 
<img width="424" alt="Screenshot 2023-03-18 at 15 40 45" src="https://user-images.githubusercontent.com/58692590/226113005-24a29fb7-51a6-4334-bb5f-df773298d734.png">